lua/hop/init.lua

```diff
@@ -102,6 +102,13 @@ local function hint_with(hint_mode, opts)
       if #line_hints.hints == 1 then
         h = line_hints.hints[1]
         unhl_and_unmark(0, hl_ns, top_line, bot_line)
+        if vim.api.nvim_get_mode().mode == 'no' then
+          if motion_is_linewise then
+            vim.cmd("normal! V")
+          else
+            vim.cmd("normal! v")
+          end
+        end
```

Comment:
This makes sure that line based motions are linewise and all others are inclusive (include the target character). The v and V keybindings are covered in https://neovim.io/doc/user/motion.html#forced-motion and there's a description about inclusive/exclusive/linewise above the linked part of the docs.
I had no idea one could use v and V like this in Vim until I saw the easymotion plugin doing this for its motions. Should I add another guard on the else so that it only makes the motions inclusive if the user opts in with a setting?

Reply:
That makes sense, but we have to ensure this doesn’t break the existing features (i.e. visual selection started in `visual` mode should stay in visual mode and not be switched in visual-line and vice versa).